Expensify/App
# [No QA] Upgrade React Native to 0.66

### Details
This pull request upgrades the following packages:

- **react-native**: upgraded from `0.64.1` to `0.66.4`
- **react-native-reanimated**: upgraded from `2.3.0-alpha.1` to `2.3.1`
    - Upgraded because it fixes https://github.com/software-mansion/react-native-reanimated/issues/2456
- **metro-react-native-babel-preset**: upgraded from `^0.61.1` to `^0.66.2`
    - Did this as it was recommended by the [React Native Upgrade Helper](https://react-native-community.github.io/upgrade-helper/?from=0.64.1&to=0.66.4)
- **react-native-config**: upgraded from `1.4.0` to `1.4.5`
    - Was necessary to get Android to build
- **react-native-plaid-link-sdk**: upgraded from `7.1.0` to `7.2.0`
    - Was necessary to get Android to build

And adds the `react-native-clean-project` as a dev dependency, along with a useful `clean` script. More details on all that does [here](https://github.com/pmadruga/react-native-clean-project#content). Note that we're using an alpha version because it fixed [this issue](https://github.com/pmadruga/react-native-clean-project/issues/79) for me.

**Note:** The [React Native Upgrade Helper](https://react-native-community.github.io/upgrade-helper/?from=0.64.1&to=0.66.4) was used to guide some of these changes.

**Note:** We should follow-up with an issue to upgrade react-native-modal because it's throwing errors due to the deprecated `removeEventListener` function. This has been fixed in their latest version.

**Note:** We should also follow-up with an issue to upgrade urbanairship-react-native, because it's throwing warnings like `new NativeEventEmitter() was called with a non-null argument without the required removeListeners method`

### Fixed Issues
Coming from https://github.com/Expensify/App/pull/6898#issuecomment-1002236004 and https://github.com/Expensify/App/issues/6874#issuecomment-1000493831

### Tests
1. (optional) Run `npm run clean` to clean the project
1. Install node modules and pods: `npm i && cd ios && pod install && cd ..`
1. Run the app
1. Run various basic regression tests to verify that the app is working as expected.

### QA Steps
None, just regression testing.

### Tested On


### Screenshots
#### Web
![image](https://user-images.githubusercontent.com/47436092/148093624-438c0172-732d-4a0c-9db9-22d37baa311c.png)

#### Mobile Web
<img src="https://user-images.githubusercontent.com/47436092/148094568-0ea0c87f-130f-49d1-bca5-903cb89fff9e.png" alt="" width="350px" />
<img src="https://user-images.githubusercontent.com/47436092/148094643-b841c846-01c5-4cca-b99a-961b15ee5361.png" alt="" width="350px" />

#### Desktop
![image](https://user-images.githubusercontent.com/47436092/148095025-925c4e3d-1a4e-4557-9ec0-ddca541f8d67.png)
![image](https://user-images.githubusercontent.com/47436092/148095068-ce07aa2d-3433-439c-ac81-dfef7a50b3bb.png)

#### iOS
<img src="https://user-images.githubusercontent.com/47436092/147703302-f8e2bd01-9f28-47ae-bf1f-c1b8fe2accba.png" alt="" width="350px" />
<img src="https://user-images.githubusercontent.com/47436092/147703370-6f848b5e-3526-48fe-a096-5fb4b51c8ef9.png" alt="" width="350px" />

#### Android
<img src="https://user-images.githubusercontent.com/47436092/147760739-6b8381a4-07d8-483c-ac1f-19fd742893d4.png" alt="" width="350px" />
<img src="https://user-images.githubusercontent.com/47436092/147760779-c4b98cd8-6a7b-4adb-8d76-ed65d8690485.png" alt="" width="350px" />
<img src="https://user-images.githubusercontent.com/47436092/147762608-9997607b-0c16-4bba-ac7b-a09d43fe4ce0.png" alt="" width="350px" />
